spark/src/main/java/org/apache/iceberg/spark/source/IcebergBatchScan.java

```diff
@@ -321,7 +302,160 @@ private Schema lazyExpectedSchema() {
     }
   }
 
-  private static class TaskDataReader implements InputPartitionReader<InternalRow> {
+  @Override
+  public InputPartition[] planInputPartitions() {
+    String tableSchemaString = SchemaParser.toJson(table.schema());
+    String expectedSchemaString = SchemaParser.toJson(lazySchema());
+
+    List<CombinedScanTask> scanTasks = tasks();
+    InputPartition[] readTasks = new InputPartition[scanTasks.size()];
+    for (int i = 0; i < scanTasks.size(); i++) {
+      readTasks[i] = new BatchReadInputPartition(scanTasks.get(i), tableSchemaString, expectedSchemaString, fileIo,
+          encryptionManager, caseSensitive);
+    }
+
+    return readTasks;
+  }
+
+  @Override
+  public IcebergRowReaderFactory createReaderFactory() {
+    return new IcebergRowReaderFactory();
+  }
+
+  private Schema lazySchema() {
+    if (schema == null) {
+      if (requestedSchema != null) {
+        this.schema = SparkSchemaUtil.prune(table.schema(), requestedSchema);
+      } else {
+        this.schema = table.schema();
+      }
+    }
+    return schema;
+  }
+
+  private StructType lazyType() {
+    if (type == null) {
+      this.type = SparkSchemaUtil.convert(lazySchema());
+    }
+    return type;
+  }
+
+  @Override
+  public StructType readSchema() {
+    return lazyType();
+  }
+
+
+  public static class PartitionRowConverter implements Function<StructLike, InternalRow> {
```

Comment:
Why was this class relocated? Relocations make it more difficult to see what has changed.